knack/arguments.py

```diff
@@ -87,63 +124,82 @@ def get_cli_argument(self, command, name):
 
 
 class ArgumentsContext(object):
-    def __init__(self, command_loader, command):
+    def __init__(self, command_loader, command_scope):
+        """ Context manager to register arguments
+
+        :param command_loader: The command loader that arguments should be registered into
+        :type command_loader: knack.commands.CLICommandsLoader
+        :param command_scope: The command scope this context manager should have
+        :type command_scope: str
+        """
         self.command_loader = command_loader
-        self.commmand = command
+        self.command_scope = command_scope
 
     def __enter__(self):
         return self
 
     def __exit__(self, exc_type, exc_val, exc_tb):
         pass
 
-    def ignore(self, argument_name):
-        self.command_loader.argument_registry.register_cli_argument(self.commmand,
-                                                                    argument_name,
-                                                                    ignore_type)
-
-    def argument(self, argument_name, arg_type=None, **kwargs):
-        self.command_loader.argument_registry.register_cli_argument(self.commmand,
-                                                                    argument_name,
+    def argument(self, argument_dest, arg_type=None, **kwargs):
+        """ Register an argument for the given command scope using a knack.arguments.CLIArgumentType
+
+        :param argument_dest: The destination argument to add this argument type to
+        :type argument_dest: str
+        :param arg_type: The argument type we are registering
+        :type arg_type: knack.arguments.CLIArgumentType
+        :param kwargs: Any argument type overrides that need to be applied
+        """
+        self.command_loader.argument_registry.register_cli_argument(self.command_scope,
+                                                                    argument_dest,
                                                                     arg_type,
                                                                     **kwargs)
 
-    def register_alias(self, argument_name, options_list, **kwargs):
-        self.command_loader.argument_registry.register_cli_argument(self.commmand,
-                                                                    argument_name,
-                                                                    None,
-                                                                    options_list=options_list,
-                                                                    **kwargs)
+    def register(self, argument_dest, options_list, **kwargs):
```

Comment:
Arguably, this isn't needed at all since the `arg_type` in `argument` is optional. Also, an option list shouldn't be needed since `argument` applies the heuristic to the `dest` value.  I would (and do) use argument consistently, whether I supply an `arg_type` or not.

Reply:
Agreed. Removed.